diff --git a/lib/sysfs.c b/lib/sysfs.c
index 51ec2bfe2..51be12db2 100644
--- a/lib/sysfs.c
+++ b/lib/sysfs.c
@@ -5,6 +5,9 @@
  * Written by Karel Zak <kzak@redhat.com>
  */
 #include <ctype.h>
+#ifdef __GNU_LIBRARY__
+#include <sys/sysmacros.h>
+#endif
 
 #include "c.h"
 #include "at.h"
diff --git a/libblkid/src/partitions/partitions.c b/libblkid/src/partitions/partitions.c
index 6e55a8f3e..4be4113b6 100644
--- a/libblkid/src/partitions/partitions.c
+++ b/libblkid/src/partitions/partitions.c
@@ -14,6 +14,9 @@
 #include <fcntl.h>
 #include <ctype.h>
 #include <sys/types.h>
+#ifdef __GNU_LIBRARY__
+#include <sys/sysmacros.h>
+#endif
 #include <sys/stat.h>
 #include <errno.h>
 #include <stdint.h>
diff --git a/libblkid/src/topology/dm.c b/libblkid/src/topology/dm.c
index e061632ca..8708215d2 100644
--- a/libblkid/src/topology/dm.c
+++ b/libblkid/src/topology/dm.c
@@ -17,6 +17,9 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#ifdef __GNU_LIBRARY__
+#include <sys/sysmacros.h>
+#endif
 #include <unistd.h>
 
 #include "topology.h"
diff --git a/libblkid/src/topology/evms.c b/libblkid/src/topology/evms.c
index 7a4fd554d..d1a413d90 100644
--- a/libblkid/src/topology/evms.c
+++ b/libblkid/src/topology/evms.c
@@ -18,6 +18,9 @@
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#ifdef __GNU_LIBRARY__
+#include <sys/sysmacros.h>
+#endif
 #include <unistd.h>
 
 #include "topology.h"
diff --git a/libblkid/src/topology/lvm.c b/libblkid/src/topology/lvm.c
index bd079d47b..c9b580ff3 100644
--- a/libblkid/src/topology/lvm.c
+++ b/libblkid/src/topology/lvm.c
@@ -17,6 +17,9 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#ifdef __GNU_LIBRARY__
+#include <sys/sysmacros.h>
+#endif
 #include <unistd.h>
 
 #include "topology.h"
diff --git a/libblkid/src/topology/md.c b/libblkid/src/topology/md.c
index 5eba94787..6a869adb3 100644
--- a/libblkid/src/topology/md.c
+++ b/libblkid/src/topology/md.c
@@ -18,6 +18,9 @@
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#ifdef __GNU_LIBRARY__
+#include <sys/sysmacros.h>
+#endif
 #include <unistd.h>
 
 #include "topology.h"
