diff --git a/libs/mpl/include/boost/mpl/aux_/integral_wrapper.hpp b/libs/mpl/include/boost/mpl/aux_/integral_wrapper.hpp
index 8748fbb..7ddc486 100644
--- a/libs/mpl/include/boost/mpl/aux_/integral_wrapper.hpp
+++ b/libs/mpl/include/boost/mpl/aux_/integral_wrapper.hpp
@@ -19,6 +19,8 @@
 #include <boost/mpl/aux_/config/static_constant.hpp>
 #include <boost/mpl/aux_/config/workaround.hpp>
 
+#include <boost/type_traits/is_enum.hpp>
+
 #include <boost/preprocessor/cat.hpp>
 
 #if !defined(AUX_WRAPPER_NAME)
@@ -29,6 +31,14 @@
 #   define AUX_WRAPPER_PARAMS(N) BOOST_MPL_AUX_NTTP_DECL(AUX_WRAPPER_VALUE_TYPE, N)
 #endif
 
+#if !defined(AUX_WRAPPER_BASE_PARAMS)
+#   define AUX_WRAPPER_BASE_PARAMS(N) AUX_WRAPPER_PARAMS(N)
+#endif
+
+#if !defined(AUX_WRAPPER_BASE_ARGS)
+#   define AUX_WRAPPER_BASE_ARGS(N) N
+#endif
+
 #if !defined(AUX_WRAPPER_INST)
 #   if BOOST_WORKAROUND(__MWERKS__, <= 0x2407)
 #       define AUX_WRAPPER_INST(value) AUX_WRAPPER_NAME< value >
@@ -37,22 +47,18 @@
 #   endif
 #endif
 
+#if !defined(AUX_WRAPPER_NEXT_PRIOR_BASE)
+#   define AUX_WRAPPER_NEXT_PRIOR_BASE BOOST_PP_CAT(AUX_WRAPPER_NAME,_next_prior_base)
+#endif
+
 BOOST_MPL_AUX_ADL_BARRIER_NAMESPACE_OPEN
 
-template< AUX_WRAPPER_PARAMS(N) >
-struct AUX_WRAPPER_NAME
-{
-    BOOST_STATIC_CONSTANT(AUX_WRAPPER_VALUE_TYPE, value = N);
-// agurt, 08/mar/03: SGI MIPSpro C++ workaround, have to #ifdef because some 
-// other compilers (e.g. MSVC) are not particulary happy about it
-#if BOOST_WORKAROUND(__EDG_VERSION__, <= 238)
-    typedef struct AUX_WRAPPER_NAME type;
-#else
-    typedef AUX_WRAPPER_NAME type;
-#endif
-    typedef AUX_WRAPPER_VALUE_TYPE value_type;
-    typedef integral_c_tag tag;
+template< AUX_WRAPPER_BASE_PARAMS(N), bool IsEnum >
+struct AUX_WRAPPER_NEXT_PRIOR_BASE;
 
+template< AUX_WRAPPER_BASE_PARAMS(N) >
+struct AUX_WRAPPER_NEXT_PRIOR_BASE< AUX_WRAPPER_BASE_ARGS(N), false >
+{
 // have to #ifdef here: some compilers don't like the 'N + 1' form (MSVC),
 // while some other don't like 'value + 1' (Borland), and some don't like
 // either
@@ -69,9 +75,31 @@ struct AUX_WRAPPER_NAME
     typedef AUX_WRAPPER_INST( BOOST_MPL_AUX_STATIC_CAST(AUX_WRAPPER_VALUE_TYPE, (N + 1)) ) next;
     typedef AUX_WRAPPER_INST( BOOST_MPL_AUX_STATIC_CAST(AUX_WRAPPER_VALUE_TYPE, (N - 1)) ) prior;
 #else
-    typedef AUX_WRAPPER_INST( BOOST_MPL_AUX_STATIC_CAST(AUX_WRAPPER_VALUE_TYPE, (value + 1)) ) next;
-    typedef AUX_WRAPPER_INST( BOOST_MPL_AUX_STATIC_CAST(AUX_WRAPPER_VALUE_TYPE, (value - 1)) ) prior;
+    typedef AUX_WRAPPER_INST( BOOST_MPL_AUX_STATIC_CAST(AUX_WRAPPER_VALUE_TYPE, (N + 1)) ) next;
+    typedef AUX_WRAPPER_INST( BOOST_MPL_AUX_STATIC_CAST(AUX_WRAPPER_VALUE_TYPE, (N - 1)) ) prior;
 #endif
+};
+
+template< AUX_WRAPPER_BASE_PARAMS(N) >
+struct AUX_WRAPPER_NEXT_PRIOR_BASE< AUX_WRAPPER_BASE_ARGS(N), true > {};
+
+template< AUX_WRAPPER_PARAMS(N) >
+struct AUX_WRAPPER_NAME
+    : AUX_WRAPPER_NEXT_PRIOR_BASE<
+            AUX_WRAPPER_BASE_ARGS(N),
+          boost::is_enum<AUX_WRAPPER_VALUE_TYPE>::value
+      >
+{
+    BOOST_STATIC_CONSTANT(AUX_WRAPPER_VALUE_TYPE, value = N);
+// agurt, 08/mar/03: SGI MIPSpro C++ workaround, have to #ifdef because some
+// other compilers (e.g. MSVC) are not particulary happy about it
+#if BOOST_WORKAROUND(__EDG_VERSION__, <= 238)
+    typedef struct AUX_WRAPPER_NAME type;
+#else
+    typedef AUX_WRAPPER_NAME type;
+#endif
+    typedef AUX_WRAPPER_VALUE_TYPE value_type;
+    typedef integral_c_tag tag;
 
     // enables uniform function call syntax for families of overloaded 
     // functions that return objects of both arithmetic ('int', 'long',
@@ -89,5 +117,8 @@ BOOST_MPL_AUX_ADL_BARRIER_NAMESPACE_CLOSE
 
 #undef AUX_WRAPPER_NAME
 #undef AUX_WRAPPER_PARAMS
+#undef AUX_WRAPPER_BASE_PARAMS
+#undef AUX_WRAPPER_BASE_ARGS
 #undef AUX_WRAPPER_INST
+#undef AUX_WRAPPER_NEXT_PRIOR_BASE
 #undef AUX_WRAPPER_VALUE_TYPE
diff --git a/libs/mpl/include/boost/mpl/integral_c.hpp b/libs/mpl/include/boost/mpl/integral_c.hpp
index 987a285..13c7c7b 100644
--- a/libs/mpl/include/boost/mpl/integral_c.hpp
+++ b/libs/mpl/include/boost/mpl/integral_c.hpp
@@ -28,6 +28,8 @@
 
 #define AUX_WRAPPER_NAME integral_c
 #define AUX_WRAPPER_VALUE_TYPE T
+#define AUX_WRAPPER_BASE_PARAMS(N) typename T, T N
+#define AUX_WRAPPER_BASE_ARGS(N) T, N
 #define AUX_WRAPPER_INST(value) AUX_WRAPPER_NAME< T, value >
 #include <boost/mpl/aux_/integral_wrapper.hpp>

